plutus-core/plutus-ir/src/PlutusIR/Transform/Inline/CallSiteInline.hs

```diff
@@ -144,13 +146,14 @@ callSiteInline processedTSize = go
         -- than it would have before, but we can make it happen more often.
         -- We could potentially do this safely in non-conservative mode.
         rhsPure <- isTermBindingPure (varStrictness varInfo) headRhs
+        thresh <- view iiInlineThreshold
         if costIsOk && rhsPure then do
           -- rename the rhs of the variable before any substitution
           renamedRhs <- rename headRhs
           applyAndBetaReduce renamedRhs args >>= \case
             Just inlined -> do
-              let -- Inline only if the size is no bigger than not inlining.
-                  sizeIsOk = termSize inlined <= processedTSize
+              let -- Inline only if the size is no bigger than not inlining plus threshold.
+                  sizeIsOk = termSize inlined <= processedTSize + fromIntegral (max 0 thresh)
```

Comment:
No `fromIntegral` would be needed if thresh has type `Size` (from the `PlutusCore.Size` module.